### DISCUSSION TITLE
Different Menus for Different Access Levels

### DISCUSSION THREAD

**Discussion by acrossan** [templaza/astroid-framework#305] [Q&A]
We have basically two different menus that we display on the site - one that appears when a user isn't logged in and a different menu when the user is logged in.

I can move the menus to a different module position and accomplish that, but I don't have an option to not show any menus in the header - I have to select one. I could hide it with css, but that seems like an ugly fix.

I could also turn off the astroid-header module, but then I don't have a way to display an offcanvas menu.

All help is appreciated.  Thanks!

**Comment by Chacapamac** [comment]
I @acrossan 
I will try to help!

You cannot use Joomla Access Rights for that in specifics menu items?
![Screen Shot 2023-03-15 at 9 03 06 AM](https://user-images.githubusercontent.com/1907276/225317120-94f1d63c-10fd-4749-91fc-8798a74493f2.jpg)

**Comment by Chacapamac** [comment]
@acrossan 
**Is this help You?**

**Comment by acrossan** [comment]
@Chacapamac Condensing all the menu items from several menus into one menu and then adjusting the permissions for each item may be an option, but that leaves the issue not displaying a navigation menu at the top of the page. I'd prefer there was another option than hiding it with CSS.

**Comment by Chacapamac** [comment]
@acrossan 
Ok, You mean that you have literally 2 menus. I was thinking that the same menu was different if user is logged.

Yes, you need at least one main menu for Joomla/Astroid to work.

> not show any menus in the header 

**Where do you want your menus?**

**• Can you give me a screenshot or a sketch or more info on what you want to achieve?**

**Comment by acrossan** [comment]
Here's the current site: XXXXXXX

Menus are in left sidebar except on mobile when menu appears via toggle.

**Comment by Chacapamac** [comment]
@acrossan 
For sure, merging the 2 menus is always the best thing to do **but this is your website.**

• The **menu in the Header** have to be the **Main Menu**
• The **menu on the side** can be done in a menu module
• To merge both menu in mobile view you need to insert the **menu on the side** in the **Main menu** in the order you want your mobile menu to appear and hide that side menu on larger devices and show it only when the Hamburger mobile menu appear 
— (On Bootstrap (MD) devices) (You will have to add the classes —> d-block d-lg-none to the hidden menu to show only in MD Device

• For menus that appear only when logged in, use Joomla Permission.

**Some Advice:**
**• Advice 1**
![Screen Shot 2023-03-22 at 3 45 02 PM](https://user-images.githubusercontent.com/1907276/227020991-35ec718e-2419-487c-af54-912fa2304265.jpg)

**Note:** Your “**Home**” menu completely disappear in “**Results Other Achieved**” for me anyway this is an interface error, especially the “Home” button should ALWAYS be at the same place and ALWAYS reachable.

**Reply by acrossan** [reply]
Where would I add the "d-block" and  "d-lg-none" classes?

**Reply by Chacapamac** [reply]
![menu](https://user-images.githubusercontent.com/1907276/227055708-42f3738e-f803-488e-a026-e7f7192f7e6f.jpg)

![Menu item](https://user-images.githubusercontent.com/1907276/227055714-770c8f3c-eac8-4584-a50e-a0de75874572.jpg)

**Just for info on theses classes given by Bootstrap 5 to show/hide elements**
—> https://getbootstrap.com/docs/5.0/utilities/display/

**• My own way to remember what is what     :)**

**Show on larger screen than**————————————————————————————
**e.g.** `<div class="d-lg-none">—> hide on lg and wider </div>`

**Show on LARGER screen than:**
XS = .d-none .d-sm-block
SM = .d-none .d-md-block
MD = .d-none .d-lg-block
LG = .d-none .d-xl-block
XL = .d-none .d-xxl-block

**Show on smaller screen than** ————————————————————————————
**e.g.** `<div class=“d-block d-lg-none”>—> show on screens smaller than lg</div>`

**Show on SMALLER screen than:**
XL = .d-block .d-xxl-none
LX = .d-block .d-xl-none
LG = .d-block .d-lg-none
MD = .d-block .d-md-none
SM = .d-block .d-sm-none

**Reply by acrossan** [reply]
In other words, I'd add the classes to each menu item. Got it.

**Reply by Chacapamac** [reply]
The menu items in the Main Menu that you want to show only on the mobile menu. 
— It will be the repeat of the secondary module “**menu on the side**”. that is in a Menu Module (not in the top Main Menu)
— You need to do that because ONLY the Main Menu appear in the Mobile Menu.

**Comment by Chacapamac** [comment]
@acrossan 
Just tell me if I can put this discussion as SOLVED

**Good Day!**

**Reply by acrossan** [reply]
You can. I don't think I'll be using this template for this project, but maybe on a different one in the future.

**Reply by Chacapamac** [reply]
No problem, anytime.
